### ISSUE TITLE
add sims_transform function

### ISSUE THREAD

**Issue by audrey-b** [poissonconsulting/sims#12]
A function that allows data transformations: example drop some values to create missing data.
Both the original and transformed data shall be saved because
-the transformed data are the one analysed
-but the original data will be used to compare an estimate to the true value (e.g. with simcases)

**Comment by joethorley** [comment]
makes sense - effectively this will be manipulations on an nlists object. Does it make most sense to do this through an R code block like mcmc_derive?

**Comment by joethorley** [comment]
I'm wondering if update.nlists() would make most sense in terms of a function?

**Comment by joethorley** [comment]
actually I think the best thing is if I add `mcmc_derive.nlists()` to the mcmcderive package
